homeassistant/components/myuplink/diagnostics.py

```diff
@@ -0,0 +1,46 @@
+"""Diagnostics support for myUplink."""
+from __future__ import annotations
+
+from typing import Any
+
+from myuplink import MyUplinkAPI
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+TO_REDACT = {"access_token", "refresh_token", "serialNumber"}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry.
+
+    Pick up fresh data from API and dump it.
+    """
+    api: MyUplinkAPI = hass.data[DOMAIN][config_entry.entry_id].api
+    myuplink_data = {}
+    myuplink_data["my_systems"] = await api.async_get_systems_json()
+    for system in myuplink_data["my_systems"]["systems"]:
+        myuplink_data["my_systems"]["devices"] = []
```

Comment:
This line can be moved up out of the loop.

Reply:
Done